crates/contract/src/lib.rs

```diff
@@ -3059,4 +3089,98 @@ mod tests {
             .build());
         contract.remove_update_vote();
     }
+
+    /// Test that `vote_update` correctly filters out non-participant votes when checking threshold.
+    ///
+    /// This is a regression test for a bug where votes from accounts that were no longer
+    /// participants (e.g., after resharing) were still counted toward the update threshold.
+    ///
+    /// The test verifies that only votes from current participants are counted:
+    /// - With threshold=2 and 3 participants, we need 2 valid participant votes
+    /// - Adding 2 non-participant votes + 1 participant vote should NOT meet threshold (returns false)
+    /// - Adding a 2nd participant vote should meet threshold (returns true)
+    #[test]
+    pub fn test_vote_update_filters_non_participant_votes() {
+        // given: a running state with 3 participants and threshold of 2
+        let running_state = gen_running_state(1);
+        let running_state = RunningContractState::new(
+            running_state.domains.clone(),
+            running_state.keyset.clone(),
+            ThresholdParameters::new(gen_participants(3), Threshold::new(2)).unwrap(),
+        );
```

Comment:
I would find it cleaner to do one of the following:
- use the output of `gen_running_state(1)` directly - we just need to loop over the participants for voting.
- not use `gen_running_state()`, as we don't really require domains or keysets